File: CS/MasterDetailProject/CustomGridControl/CustomGridViewRegistrator.cs
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Views.Grid;
using MasterDetailProject.CustomGridControl;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MasterDetailProject {
    class CustomGridViewRegistrator : GridInfoRegistrator {

        public override string ViewName {
            get {
                return "CustomGridView";
            }
        }

        public override DevExpress.XtraGrid.Views.Base.BaseView CreateView(DevExpress.XtraGrid.GridControl grid) {
            return new CustomGridView(grid as GridControl);
        }

        public override DevExpress.XtraGrid.Views.Base.ViewInfo.BaseViewInfo CreateViewInfo(DevExpress.XtraGrid.Views.Base.BaseView view) {
            return new CustomGridViewInfo((GridView)view);
        }

    }
}

File: VB/MasterDetailProject/CustomGridControl/CustomGridViewRegistrator.vb
Imports Microsoft.VisualBasic
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Views.Grid
Imports MasterDetailProject.CustomGridControl
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace MasterDetailProject
	Friend Class CustomGridViewRegistrator
		Inherits GridInfoRegistrator

		Public Overrides ReadOnly Property ViewName() As String
			Get
				Return "CustomGridView"
			End Get
		End Property

		Public Overrides Function CreateView(ByVal grid As DevExpress.XtraGrid.GridControl) As DevExpress.XtraGrid.Views.Base.BaseView
			Return New CustomGridView(TryCast(grid, GridControl))
		End Function

		Public Overrides Function CreateViewInfo(ByVal view As DevExpress.XtraGrid.Views.Base.BaseView) As DevExpress.XtraGrid.Views.Base.ViewInfo.BaseViewInfo
			Return New CustomGridViewInfo(CType(view, GridView))
		End Function

	End Class
End Namespace
